In [1]:
data_train<-read.csv(choose.files())

In [2]:
data_train[1:5,]

free credit money lab meeting conference Spam
1 0.00 0      0     0   0       0.00       0   
2 1.23 0      0     0   0       0.00       1   
3 0.00 0      0     0   0       0.00       0   
4 0.00 0      0     0   0       0.00       1   
5 0.00 0      0     0   0       0.77       0

In [4]:
attach(data_train)
model1<-glm(Spam~free+credit+money+lab+meeting+conference, family = binomial)

The following objects are masked from data_train (pos = 3):

    conference, credit, free, lab, meeting, money, Spam

Warning message:
"glm.fit: fitted probabilities numerically 0 or 1 occurred"

In [5]:
summary(model1)


Call:
glm(formula = Spam ~ free + credit + money + lab + meeting + 
    conference, family = binomial)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-6.5139  -0.8196  -0.1856   0.6396   2.5178  

Coefficients:
            Estimate Std. Error z value Pr(>|z|)    
(Intercept)  -0.9185     0.0919  -9.994  < 2e-16 ***
free          1.1067     0.1781   6.213 5.20e-10 ***
credit        7.7522     1.9152   4.048 5.17e-05 ***
money         7.1476     1.1158   6.406 1.50e-10 ***
lab          -5.8049     2.4464  -2.373   0.0177 *  
meeting      -6.1952     2.9551  -2.096   0.0360 *  
conference   -4.6155     2.8880  -1.598   0.1100    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 1348.42  on 999  degrees of freedom
Residual deviance:  913.29  on 993  degrees of freedom
AIC: 927.29

Number of Fisher Scoring iterations: 10


Meeting turns out be significant, but it is negatively correlated with spam. So, the word meeting is more of a genuine email.
The co-efficient of money is 7.1476. If the email contains the word money, the odds of spam are exp(7.1476)=1271 times higher.
-0.9185 – 6.1952 + 14.2952 = 7.1815 where p is the probability (Probability = 1/[1+exp(-7.1815)]) = 0.99924

In [7]:
data1<-read.csv(choose.files())
predicted<-predict(model1, data1, type = 'response')
threshold<-0.5
predf<-cut(predicted, breaks = c(-Inf, threshold, +Inf), labels = c("0","1"))
ctab<-table(data1$Spam, predf, dnn = c("actual","predicted"))
addmargins(ctab)

predicted
actual 0   1   Sum
   0   298  13 311
   1    69 120 189
   Sum 367 133 500

Accuracy = (298+120)/500 = 83.6%
Error = 1-Accuracy = 16.4%
As the error rate is high, model is not good.